src/queries/role.js

```diff
@@ -56,6 +58,11 @@ function queryAllByRole(
           }) === false
         : true
     })
+    .filter(element => {
+      return typeof name === 'string'
+        ? computeAccessibleName(element) === name
```

Comment:
Could we make the `name` option support everything that `TextMatch` supports? Like function and regex?

Reply:
I'd prefer to keep the initial feature as simple as possible. I already have a hard time picturing use cases for `getByRole(/regex/)`. Just because we can do it doesn't mean we should. Especially since we have to teach a new option and all the overloads for it.

It's very likely a feature that will be requested and should be available at some point. Just not from the start and without a non-technical justification

One goal of this feature was that you don't need to use regular expressions to properly ignore certain whitespace (which is already handled by `computeAccessibleName`).

Reply:
I think he meant `getByRole('button', {name: /xyz/})`

Reply:
I know. I'd like to see actual use cases first though since we don't parse text with regular expressions. I'm not opposed to this feature per se. Just want to keep the PR as small as possible since I'd like to keep the amount of work I have to put into features I don't need as small as possible.

Reply:
Implemented.